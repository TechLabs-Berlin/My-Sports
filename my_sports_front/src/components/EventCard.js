import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';
import StarRatings from 'react-star-ratings';

export default function EventCard({ event }) {
    return (
        <div className="form-box-white thick-bottom-border">
            <div className="card" style={{width: 100 + '%'}}>
                <div className="card-body">
                    <h3 className="event-page-event-title card-title">{event.title}</h3>
                    <div className={`bottom-border-color-${event.sport} row justify-content-space-between`}>
                        <div className="col">
                            <p className="card-text text-muted">{moment(event.date).format("dddd, MMM Do")}</p>
                            <p className="card-text text-muted">{event.location}</p>
                            {event.court &&  <p className="card-text text-muted">{event.court}</p>}
                        </div>
                            <div className="col">
                            <img className="event-page-sport-image" src={`../images/sport-images/${event.sport}.png`} />
                        </div>
                    </div>
                    <div className="row">
                        <p id="dashboard-player" className="col card-text text-muted player-count-text">Players: <span className="error-text">{event.teammates.length}/{event.size}</span></p>
                        <p id="dashboard-stars" className="col dashboard-stars"><StarRatings
                            rating={event.skill / event.teammates.length}
                            starRatedColor="#E9B467"
                            numberOfStars={5}
                            starDimension="1.2em"
                            starSpacing="0.7px"
                            /></p>
                    </div>
                    <div className="accordion" id={"accordionExample" + event._id}>
                        <div className="accordion-item">
                            <h2 className="accordion-header" id={"heading" + event._id}>
                                <button className="accordion-button" type="button" data-bs-toggle="collapse" data-bs-target={"#collapse" + event._id} aria-expanded="true" aria-controls={"collapse"+ event._id}></button>
                            </h2>
                            <div id={"collapse"+ event._id} className="accordion-collapse collapse" aria-labelledby={"heading"+ event._id} data-bs-parent={"#accordionExample"+ event._id}>
                                <div className="accordion-body">
                                    <div className="row justify-content-center"><button className="edit-event-button"><Link className="edit-event-link" to={"/event/" + event._id}>View Event</Link></button></div>
                                </div>
                            </div>
                        </div>
                    </div>
                </div>
            </div>
        </div>
    )
}
