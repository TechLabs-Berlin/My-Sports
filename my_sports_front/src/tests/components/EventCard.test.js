import React from "react";
import { shallow } from "enzyme";
import EventCard from "../../components/EventCard";
import moment from "moment";
import "../setupTests";

const event = {
  title: "",
  sport: "",
  size: 0,
  date: moment(),
  time: moment(),
  location: "mitte",
  host: "",
  about: "",
  court: "",
  teammates: "",
};

test("should render event card correctly", () => {
  const wrapper = shallow(<EventCard event={event} />);
  expect(wrapper.debug()).toMatchSnapshot();
});
