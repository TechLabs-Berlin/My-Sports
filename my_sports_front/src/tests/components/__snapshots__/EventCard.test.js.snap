// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render event card correctly 1`] = `
"<div className=\\"form-box-white thick-bottom-border\\">
  <div className=\\"card\\" style={{...}}>
    <div className=\\"card-body\\">
      <h3 className=\\"event-page-event-title card-title\\" />
      <div className=\\"bottom-border-color- row justify-content-space-between\\">
        <div className=\\"col\\">
          <p className=\\"card-text text-muted\\">
            Thursday, Jan 1st
          </p>
          <p className=\\"card-text text-muted\\">
            mitte
          </p>
        </div>
        <div className=\\"col\\">
          <img className=\\"event-page-sport-image\\" src=\\"../images/sport-images/.png\\" />
        </div>
      </div>
      <div className=\\"row\\">
        <p id=\\"dashboard-player\\" className=\\"col card-text text-muted player-count-text\\">
          Players: 
          <span className=\\"error-text\\">
            0
            /
            0
          </span>
        </p>
        <p id=\\"dashboard-stars\\" className=\\"col dashboard-stars\\">
          <StarRatings rating={NaN} starRatedColor=\\"#E9B467\\" numberOfStars={5} starDimension=\\"1.2em\\" starSpacing=\\"0.7px\\" typeOfWidget=\\"Star\\" changeRating={{...}} starHoverColor=\\"rgb(230, 67, 47)\\" starEmptyColor=\\"rgb(203, 211, 227)\\" gradientPathName=\\"\\" ignoreInlineStyles={false} svgIconPath=\\"m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z\\" svgIconViewBox=\\"0 0 51 48\\" />
        </p>
      </div>
      <div className=\\"accordion\\" id=\\"accordionExampleundefined\\">
        <div className=\\"accordion-item\\">
          <h2 className=\\"accordion-header\\" id=\\"headingundefined\\">
            <button className=\\"accordion-button\\" type=\\"button\\" data-bs-toggle=\\"collapse\\" data-bs-target=\\"#collapseundefined\\" aria-expanded=\\"true\\" aria-controls=\\"collapseundefined\\" />
          </h2>
          <div id=\\"collapseundefined\\" className=\\"accordion-collapse collapse\\" aria-labelledby=\\"headingundefined\\" data-bs-parent=\\"#accordionExampleundefined\\">
            <div className=\\"accordion-body\\">
              <div className=\\"row justify-content-center\\">
                <button className=\\"edit-event-button\\">
                  <Link className=\\"edit-event-link\\" to=\\"/event/undefined\\">
                    View Event
                  </Link>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>"
`;
